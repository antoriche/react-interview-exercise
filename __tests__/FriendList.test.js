import React from 'react';
import FriendList from '../src/components/FriendList';
import renderer from 'react-test-renderer';

describe('FriendList', ()=>{
  it('render correctly', () => {
    const component = renderer.create(
      <FriendList
          friends = {[{name: "Michel", gender: 'M', starred: false}, {name: "Dominique", gender: 'F', starred: true}]}
          actions = {{starFriend:()=>{}}}
      />,
    );
    
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});