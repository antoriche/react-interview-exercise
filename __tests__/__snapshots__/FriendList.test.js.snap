exports[`FriendList render correctly 1`] = `
<ul
  className={undefined}>
  <li
    className={undefined}>
    <div
      className={undefined}>
      <div>
        <i
          className="fa fa-mars" />
        <span
          className={undefined}>
          Michel
        </span>
      </div>
      <div>
        <small>
          xx friends in common
        </small>
      </div>
    </div>
    <div
      className={undefined}>
      <button
        className="btn btn-default undefined"
        onClick={[Function]}>
        <i
          className="fa fa-star-o" />
      </button>
      <button
        className="btn btn-default undefined"
        onClick={[Function]}>
        <i
          className="fa fa-trash" />
      </button>
    </div>
  </li>
  <li
    className={undefined}>
    <div
      className={undefined}>
      <div>
        <i
          className="fa fa-venus" />
        <span
          className={undefined}>
          Dominique
        </span>
      </div>
      <div>
        <small>
          xx friends in common
        </small>
      </div>
    </div>
    <div
      className={undefined}>
      <button
        className="btn btn-default undefined"
        onClick={[Function]}>
        <i
          className="fa fa-star" />
      </button>
      <button
        className="btn btn-default undefined"
        onClick={[Function]}>
        <i
          className="fa fa-trash" />
      </button>
    </div>
  </li>
</ul>
`;
